src/Menu/Menu.component.tsx

```diff
@@ -3,119 +3,147 @@ import * as React from 'react';
 // VENDOR
 import classNames from 'classnames';
 import styled, { css } from '@xstyled/styled-components';
-import { th } from '@xstyled/system';
+import { th, variant } from '@xstyled/system';
 import { lighten } from 'polished';
-// COMPONENTS
+// ANCHOR
 import { ItemProps } from './Item/Item.component';
+import { space as spaceStyles, SpaceProps } from '@xstyled/system';
 
-interface MenuProps extends React.HTMLAttributes<HTMLDivElement> {
+enum MenuSizes {
+    sm = 'sm',
+    md = 'md',
+    lg = 'lg',
+}
+
+export type MenuSizeProps = MenuSizes | 'sm' | 'md' | 'lg';
```

Comment:
I'd rather use the Typescript tool for solving this while also allowing pure JS persons to use strings than try to use an approach that doesn't give the advantages of either, but I have been doing some digging and by using the Enum it may be the case that the transpiled output isn't unlik what you're positing above, but instead of requiring an object's key, it'll accept the value of the keys in the JS version which would be a string. I'd need to build out a candidate to test this, but it's what I'm hoping for. That way TS gets an Enum and JS gets a string without any configuration.

Reply:
I definitely can appreciate it using the value of the key rather than the key itself. I've been trying to find an approach that lets us not have to define the values for `type`'s twice, and also lets us export those values easily for use, such as for knobs.